### PR TITLE
[Stats] Make aggeration window and report  interval configurable.

### PR DESCRIPTION
**Pull request by ashione**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Comment by ashione** [comment]
@rkooo567 @edoakes This PR based on last metric exporter client focuses on making harvest and report interval configurable. I'll merge this lastest master if last PR is merged.

**Comment by edoakes** [comment]
@ashione please merge latest master when you have the chance!

File: src/ray/stats/metric.h [hunk 1]
```diff
@@ -43,6 +43,15 @@ class StatsConfig final {
...
   bool IsStatsDisabled() const;
 
+  void SetReportInterval(const absl::Duration interval);
+
+  const absl::Duration &GetReportInterval() const;
+
+  void SetHarvestInterval(const absl::Duration interval);
+
+  const absl::Duration &GetHarvestInterval() const;
+  absl::Duration report_interval = absl::Seconds(1);
```
Comment (rkooo567) [inline src/ray/stats/metric.h:53]: Duplicated?

File: src/ray/stats/metric.h [hunk 2]
```diff
@@ -52,6 +61,8 @@ class StatsConfig final {
  private:
   TagsType global_tags_;
   bool is_stats_disabled_ = true;
+  absl::Duration report_interval_ = absl::Seconds(10);
```
Comment (rkooo567) [inline src/ray/stats/metric.h:64]: Can you write description for both `report_interval_` and `harvest_interval`?

**Review by rkooo567 — COMMENTED** [review]
LGTM.

**Review by rkooo567 — APPROVED** [review]

File: src/ray/stats/metric.h (hunk 1 above)
Comment (ashione) [reply src/ray/stats/metric.h]: > Duplicated?

Duplicated definition fixed.

**Comment by ashione** [comment]
> @ashione please merge latest master when you have the chance!

This pr has merged lastest master. Any comments else?

**Comment by rkooo567** [comment]
I think if you merge this commit, test failures will be likely to be fixed. bfb8a28a3cca940621841a52854ab3dd8ef37a76 

Can you merge the latest master one more time? We'll merge it after that. Please feel free to create the next PR.

**Comment by rkooo567** [comment]
@edoakes Can we merge this PR?

**Comment by rkooo567** [comment]
Will merge this by tomorrow.

**Comment by rkooo567** [comment]
@raulchen Can you merge this PR?

**Pull request merged**